src/main/java/org/openpnp/gui/GcodeDriverConsole.java

```diff
@@ -0,0 +1,200 @@
+import org.openpnp.gui.support.MessageBoxes;
+import org.openpnp.machine.reference.driver.GcodeDriver;
+import org.openpnp.machine.reference.driver.GcodeDriver.Command;
+import org.openpnp.machine.reference.driver.GcodeDriver.CommandType;
+import org.openpnp.model.Configuration;
+import org.openpnp.model.LengthUnit;
+
+import com.jgoodies.forms.layout.ColumnSpec;
+import com.jgoodies.forms.layout.FormLayout;
+import com.jgoodies.forms.layout.FormSpecs;
+import com.jgoodies.forms.layout.RowSpec;
+import javax.swing.border.EtchedBorder;
+import java.awt.Color;
+import javax.swing.JTextField;
+import org.pmw.tinylog.Logger;
+
+public class GcodeDriverConsole extends AbstractConfigurationWizard {
+    private final GcodeDriver driver;
+
+    public GcodeDriverConsole(GcodeDriver driver) {
+        this.driver = driver;
+        
+        historyLen = 0;
+        historyMaxLen = 50;
+        history = new String[historyMaxLen];
+        historyCursor = 0;
+        
+        JPanel gcodeConsole = new JPanel();
+        gcodeConsole.setBorder(
+          new TitledBorder(new EtchedBorder(EtchedBorder.LOWERED, null, null), 
+            "Gcode console",TitledBorder.LEADING, 
+            TitledBorder.TOP, 
+            null, 
+            new Color(0, 0, 0)
+          )
+        );
+        contentPanel.add(gcodeConsole);
+        
+        gcodeConsole.setLayout(new FormLayout(
+                new ColumnSpec[] {
+                    FormSpecs.RELATED_GAP_COLSPEC, FormSpecs.DEFAULT_COLSPEC,
+                    FormSpecs.RELATED_GAP_COLSPEC, ColumnSpec.decode("default:grow"),
+                    FormSpecs.RELATED_GAP_COLSPEC, FormSpecs.DEFAULT_COLSPEC,
+                },
+                new RowSpec[] {
+                    FormSpecs.RELATED_GAP_ROWSPEC, RowSpec.decode("default:grow"),
+                    FormSpecs.RELATED_GAP_ROWSPEC, FormSpecs.DEFAULT_ROWSPEC,
+                }
+            )
+        );
+        
+        JScrollPane scrollPane = new JScrollPane();
+        gcodeConsole.add(scrollPane, "2, 2, 5, 1, fill, fill");
+        
+        textAreaConsole = new JTextArea();
+        textAreaConsole.setFont(new Font("Monospaced", Font.PLAIN, 13));
+        textAreaConsole.setEditable(false);
+        textAreaConsole.setLineWrap(true);
+        textAreaConsole.setRows(5);
+        scrollPane.setViewportView(textAreaConsole);
+        
+        lblCmdLine = new JLabel("Command line:");
+        gcodeConsole.add(lblCmdLine, "2, 4");
+        
+        cmdLineTextField = new JTextField();              Logger.debug("keyCode = " + e.getKeyCode() + ", historyCursor = " + historyCursor);
+
+        gcodeConsole.add(cmdLineTextField,"4, 4");
+        
+        cmdLineTextField.addKeyListener(new KeyAdapter() {
+            @Override
+            public void keyPressed(KeyEvent e) {
+            
+              switch (e.getKeyCode()) {
+                case 0:  return;
+                
+                case 38: // arrow key UP, older history
```

Comment:
Instead of hardcoding key codes, please use the constants defined on KeyEvent. See, for instance: https://docs.oracle.com/javase/7/docs/api/java/awt/event/KeyEvent.html#VK_UP